### DISCUSSION TITLE
Feature suggestions for Copilot Plus mode

### DISCUSSION THREAD

**Discussion by logancyang** [logancyang/obsidian-copilot#726] [Polls]
I'm rolling out Copilot Plus to alpha testers (priority will be given to project sponsors).

The main idea is to enable tool use and agentic RAG. You can find out more here: https://obsidiancopilot.com

I listed some features. If you would like to suggest features, please comment below!

**Comment by dinergie** [comment]
make the copilot summarize all the pdf I have in my vault, on right click, multi-selection 

set custom action copilot can do with right click on document

make copilot propose an excalidraw diagram drawing

make copilot create file(s) in obsidian, based on the prompt

**Reply by LeCheenaX** [reply]
excalidraw has its own AI tool to generate diagram so this one may be skipeed

**Comment by sector101010** [comment]
Great news. I'd like to see, from high to low:
- note creation, deletion and refactoring based on chat
- note moving and folder creation, deletion
- spaced repetition: ask me multiple choice questions on some of my notes, remember answers and ask again at appropriate time
- task management: show me (..) of my tasks dropdown where I can change their status
- canvas creation and refactoring
- side-by-side chats with two different models, answering the same questions
- mass note renaming

**Reply by TomB15** [reply]
Following are high also on my list

`
- note creation, deletion and refactoring based on chat
- note moving and folder creation, deletion 
- spaced repetition: ask me multiple choice questions on some of my notes, remember answers and ask again at appropriate time

`

**Comment by Fffrankoo** [comment]
Awesome! Other features that I think would be great, from high to low:
-	insert individual chat answers into existing notes (append to note / replace / insert at cursor position)
-	full support for local LLMs from LM Studio and Ollama for chatting and vault indexing
-	flexible context definition in chat window (current note, links to notes, all notes in a folder+subfolders, all notes with specific tags or front matter properties)
-	suggest and insert tags (preferably in front matter) based on content (giving existing tags in vault higher relevance)
-	suggest vault folder locations and move to folder based on content (giving existing folders in vault higher relevance)
-	access to note properties in front matter for context selection purposes and note refactoring
-	UI to concatenate multiple AI actions into one procedure that can be stored for re-use (library of default actions?)
-	automatic temporary note backup mechanism when existing notes are modified by the plugin (so that previous version of a note is still available or action can be rolled back)
-	UI to instruct Copilot Plus which LLM provider/LLM to use for a particular chat question

**Reply by logancyang** [reply]
Thanks for the suggestion. As for the first 3
- Insert at cursor already exists
- full support for Ollama for chatting and indexing already exists (LM studio for indexing can be improved tho)
- this is custom prompt in chat that already exists. Referencing notes with N degrees of connection can be added tho.

If this isn't what you meant please let me know.

**Reply by Fffrankoo** [reply]
Hi Logan,
Re. 1: I was thinking about how to make this a frictionless as possible for the user. Maybe just add an extra icon Append to note next to Insert at cursor.
Re. 3: Correct – I did not realize that all context selection options available in custom prompts could also be used in the chat windows (without applying a custom prompt)

**Comment by kannib** [comment]
It would be amazing to have a feature in your LLM tool that links and maps every sentence or paragraph directly to its exact spot in a PDF (currently Acrobat AI does it). The tool could highlight the referenced text in the original document, making it super easy to double-check quotes or references without wasting time searching manually. This would be a game-changer for anyone working with long documents.

**Comment by igorsancar** [comment]
1) I thought of a system to acquire knowledge faster: I often save videos and links that I find on the internet to consume later when I have time.

Sometimes the knowledge contained there is relevant, partly relevant or completely known.

The problem is that we have to take the time to consume all this content and validate it internally in our minds.

With the proposal of the second brain, by putting all our knowledge into it, an artificial intelligence could easily consume this content, creating study notes only with the knowledge that we do not yet have.

Thus, we could upload it to the main brain later and authorize the indexing of the material.



**In my suggestion, are there currently plugin that do this?**

**Comment by saltymangos** [comment]
I'm just now looking into adding AI to my Obsidian workflow and it seems clear that Copilot is near the top for current options. I like where the tool is headed.

The big feature I am seeing that I would love though is the ability to have Copilot edit the document for us with edits that we have accepted. Right now this is able to be done with a plugin call Smart Composer and it does it very well so far in my little testing. It seems like this could easily be added to Copilot to some degree by default or even as a specific @ call(@edit) if a specific workflow needed to be called to handle the editing of the document.

I will also say the way that Smart Composer handles the chat window in regards to sources, the edited section window, and inserting sourced doc content seems better implemented. Would love to see something like that in Copilot Plus or future Copilot releases.

Thinking on it now it would also be interesting to define custom workflows in the plugin as well. They could be as simple as chaining prompts together and then define that as a workflow. Would cut out having to run multiple prompts or actions on repeated content and easy ways to ensure output is more in line with what we expect.

For those that like quick bullets:

Ability to edit documents in chat and apply them once confirmed similar to how the plugin Smart Composer currently works.
Include relevant source content in chat when able.
Custom workflows even if just simply chaining prompts together.
On a side note I saw your video and it stated supporters were getting an early key for Copilot Plus and I'm curious as to what that means and if keys are still available.

**Reply by logancyang** [reply]
Thanks for the suggestion. Inline features will come in gradually after the official launch, like autocomplete, inline editing, and batch editing via agent. 

Alpha test is currently at limit but we are very close to the official launch! Stay tuned!

**Reply by saltymangos** [reply]
Appreciate the quick reply. Looking forward to tracking the progress of this plugin. Best of luck.

**Comment by shayonpal** [comment]
One of the biggest challenges I face in my vault is organization, particularly in managing proper YAML metadata. Currently, I rely on Cursor IDE to create new notes. It efficiently takes note input from me, formats it correctly, and automatically adds YAML fields and values based on my typical vault usage. While this works well, I would greatly appreciate a similar experience within Obsidian.

Importantly, iPad support is a must for me since it serves as my primary computer.

Furthermore, I utilize Actions for Obsidian to automate tasks within my vault using Apple Shortcuts. If I could create similar actions for Copilot, I could share the URL of an API documentation via shortcuts, and Copilot could automatically parse the page, generate a cheat sheet, organize the YAML data, and save it at an appropriate location. This would be an incredible feature! Currently, I accomplish most of these tasks through quite complex multi-step Shortcuts.

**Comment by seancrwhite** [comment]
Any plan to add local recording/video parsing? As it currently stands I could upload things to YouTube as private and then add the link, but it would be nice to be able to skip that step.

**Comment by sangsangfroydfroyd** [comment]
Is there a way to import over the vector data to places like Postgres Pinecone or Supabase? 
I thought about transferring over the embedded model in the .obsidian folder but I'm unsure if it's the correct format for supabase RAG
I figured it would work since the embedding would be done with the same model in n8n but am unsure 

If not another option to sync and update the vector data over to be used with n8n or Make agents would be so awesome

**Reply by sangsangfroydfroyd** [reply]
Hoping to make some sort of brainstorming agent that allows me to have a "normal conversation" with agent and it creates notes does research and creates an execution plan, updates and adds files via s3 bucket sync

**Comment by josh-clarke** [comment]
I would love to be able to open a new chat in a new tab so that I do not lose the first chat, or at least a way to save a chat so that it can be re-opened in the chat window and continued.

**Reply by logancyang** [reply]
> I would love to be able to open a new chat in a new tab so that I do not lose the first chat, or at least a way to save a chat so that it can be re-opened in the chat window and continued.

You can save the chat to a note and load it with a command.

**Reply by josh-clarke** [reply]
Oh wow, I didn't realize that. Thank you!

**Comment by marcryan** [comment]
Hi Logan, This thread looks stale but I'm going to pop this here since it seems like the best place for feature requests.

I'm a new "Believer" sub and love the work you're doing with the tool. Off the bat the one feature that would be a game changer for me is to flip the chat window/sidebar.  

What I mean is to have the text input at the top of the window and responses in descending order from the top-newest to bottom-oldest. 

I know you modeled off of Cursor and it's an equivalent gripe I have with Cursor. 

For context I work on a 34" screen with a 21:9 aspect ratio. When the chat is in the sidebar on the right of my screen I'm looking significantly to the right and down when I'm in chat mode. I know I can move the chat dialog into the main editor as a tab but that doesn't alleviate the problem of constantly looking at the bottom of the screen. 

Another challenge for us folks on a Windows machine is that any desktop notifications (toast) show up in the bottom right of the screen which covers the chat dialog. This can be particularly annoying when in the middle of a conversation since the notification always renders on the top layer of the display. 

Since I'd like to spend more time in Copilot as the primary interface for LLM conversations I'd prefer to have it eye-level. Hopefully this is simple CSS work and a toggle in settings. 

Thanks!!!

**Comment by SN-2187kk** [comment]
I think would be great if you add **local LLM support** for any implementation of Copilot Plus features (specifically for **word and sentence autocomplete**). It would be an optimized approach, since local LLMs have lower latency for these cases compared to the API approach currently in use. Using local LLMs — this is a modern practice for solving such tasks. This approach is used, for example, by JetBrains in their IDE for autocomplete. Using an API for these tasks, I consider it unnecessary.
